### PR TITLE
docs: upgrade snaplet + supabase docs.

### PR DESCRIPTION
**Pull request by khaya-zulu**
## What kind of change does this PR introduce?

Snaplet integration docs update

## Additional context

https://github.com/supabase/supabase/assets/39437696/88d4504e-d925-4349-8511-320d9df3568e

### CODE REVIEW

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 1]
```diff
@@ -7,104 +7,156 @@ export const meta = {
...
 }
 
-This step-by-step guide explains how to use Snaplet to clone your production Supabase project into another development database.
+This guide will help you get a development Supabase instance set up using Snaplet. In 15 minutes or less, you will create a snapshot of your production Supabase instance and restore it to a development environment you can safely code against.
 
-[Snaplet](https://snaplet.dev/) is a developer tool that copies a Postgres database, transforming personal information, so that you can **safely code against actual data.** This functionality makes it possible to easily achieve environment parity in Supabase.
+<Admonition type="note">
 
-Let's get started!
+If you experience any issues with getting Supabase working with Snaplet, or with this guide, feel free to come chat with us on [Discord](https://app.snaplet.dev/chat).
```
Comment (gregnr) [inline apps/docs/pages/guides/integrations/snaplet.mdx:14]:
```suggestion
If you experience any issues getting Supabase working with Snaplet, feel free to chat with the Snaplet team on [Discord](https://app.snaplet.dev/chat).
```

**Review by gregnr — COMMENTED** [review]
Noticing this update is a close copy of https://docs.snaplet.dev/tutorials/supabase-clone-environments/.

That's okay, but ideally can we try to adjust the language to be a bit more objective or from the Supabase perspective (removing things like "we at Snaptlet", etc where possible so that we don't confuse our users).

Thanks!

**Comment by khaya-zulu** [comment]
> Noticing this update is a close copy of https://docs.snaplet.dev/tutorials/supabase-clone-environments/.
> 
> That's okay, but ideally can we try to adjust the language to be a bit more objective or from the Supabase perspective (removing things like "we at Snaptlet", etc where possible so that we don't confuse our users).
> 
> Thanks!

Thanks @gregnr for the feedback, I've made a few improvements to the docs, per your suggestions.

**Comment by charislam** [comment]
Hi @khaya-zulu, thanks for the contribution!

Can you double-check that this [builds locally](https://github.com/supabase/supabase/blob/master/apps/docs/README.md) for you? The preview is throwing an error on the added page. If you need help getting local dev set up to check, give me a shout!

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+> **Highly Recommended:** Create a read-only role for your connection string. For more info on how to create a read-only role across all schemas, you can check out the [Snaplet docs](https://docs.snaplet.dev/guides/postgresql#create-a-read-only-role).
+
+### Step 1: Connect your source database (Production Database) [#connect-source-db]
+
+The first thing you’ll want to do is navigate to [https://www.snaplet.dev/](https://www.snaplet.dev/) and sign up for a new account (it’s free). Once you have successfully signed up for a new account, you’ll begin the onboarding process...
+
+![Onboarding start](/docs/img/guides/integrations/snaplet/onboarding_start.webp)
+
+On the “Connect database” step, click on "Connect Supabase" to connect your supabase account to snaplet. Proceed through all the steps of authorizing your Supabase organization then selecting Supabase project and giving Snaplet the password associated to your project.
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:38]:
```suggestion
On the “Connect database” step, click on "Connect Supabase" to connect your Supabase account to snaplet. Authorize your Supabase organization, then select your Supabase project, and enter your project password.
```

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+![Validating credentials](/docs/img/guides/integrations/snaplet/connect_to_supabase.webp)
+
+### Step 2: Create a snapshot [#create-snapshot]
+
+Once that is done, continue to the next steps:
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:44]:
```suggestion
Optionally, you can alter the data in your snapshot:
```

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+- Subsetting (skipping will create a snapshot with all your data).
+- Transforming (skipping will leave all your table columns untouched).
+
+Once you have gone through those steps, a new snapshot process will start and you will now be on the “capture step.” Wait for it to finish - this is the snapshot you will restore into your **target database.**
+
+### Step 3: Create a target database on Supabase (Development Database) [#create-target-db]
+
+Your target database is where you want Snaplet to restore the captured snapshot of your production project. This would most likely be either your staging or development Supabase project.
+
+If you don’t already have a development database setup on Supabase, you’ll need to create a new database by setting up a new project on Supabase. To create a new project you’ll have to do the following:
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:55]:
```suggestion
If you don’t already have a development database setup on Supabase, you’ll need to create a new database by setting up a new project on Supabase. To create a new project:
```

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+1. Go to [app.supabase.io](https://app.supabase.io/)
+
+2. Go to your **organization > All projects**
+
+3. In the top left click on your current project name and new project:
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:61]:
```suggestion
3. In the menu bar, open the project dropdown and click on 'New project`:
```

Avoid using directional descriptors as much as possible, so the docs stay robust if UI shifts around

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+4. Paste the access token in the terminal.
+
+### Step 5: Config setup [#config-setup]
+
+You're now ready to restore your production snapshot into your Supabase development project.
+
+1. Navigate to your project directory
+
+2. Run `snaplet config setup`
+   i. You will be shown a warning to write create `.snaplet/config` in your project directory. Select yes (y) to create the file.
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:82]: Let's stick with Arabic numerals for all lists, including sublists

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+### Things you'll need before you begin: [#prerequisite]
+
+1. **A production Supabase project’s connection string:** This can be found in Supabase via **Organization > All Projects > Your Project > Project Settings > Connection Pooling > Connection String**. We’ll refer to this in the guide as your “source database” - where your data comes from.
+
+![Supabase connection string](/docs/img/guides/integrations/snaplet/connection_string.webp)
+
+2. A development Supabase project’s connection string: Same steps as above, but a different project/environment. We’ll refer to this in the guide as your “target database” - where we’re restoring the snapshot of the source database to. If you haven’t got this set up yet, we’ll take you through the process of setting this up.
+
+> **Highly Recommended:** Create a read-only role for your connection string. For more info on how to create a read-only role across all schemas, you can check out the [Snaplet docs](https://docs.snaplet.dev/guides/postgresql#create-a-read-only-role).
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:30]: Can you convert blockquotes to [admonitions](https://github.com/supabase/supabase/blob/6e1620a20adc22ae3f32ca67a5931f4da4d93344/packages/ui/src/components/Admonition/index.tsx#L5)?

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+With all the above steps complete, we can now restore!
+
+1. Run `snaplet snapshot restore --no-reset`
+
+> **Note on `--no-reset`:** This will skip the “reset” step of the restore command. So no existing schemas will be dropped. Learn more about data operations [here](https://docs.snaplet.dev/getting-started/restoring#opting-out)
+
+## All done!
+
+If you want to learn more about Snaplet, you can explore our docs. If you have any questions, feel free to reach out on [Discord](https://app.snaplet.dev/chat).
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:99]: This sounds like it was copied over from the Snaplet docs. Can you make it more third-party? 'Our' docs here would refer to Supabase, which I'm guessing is not what you want :D

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+<div>
+
+<img src="/docs/img/guides/integrations/snaplet/snappy-with-supabase-ball.svg/" alt="Supabase connection string" width="30%" style={{ margin: "auto" }} />
+
+</div>
+
+## Supabase local development [#local-development]
+
+Supabase allows you [develop locally](/docs/guides/getting-started/local-development) through their CLI, you can easily restore any snapshot created in Snaplet to your local instance, just by running:
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:111]:
```suggestion
Supabase allows you [develop locally](/docs/guides/getting-started/local-development) through the Supabase CLI. You can restore any snapshot created in Snaplet to your local instance by running:
```

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+# Login into the Supabase CLI
+supabase login
+
+# Initialize Supabase to set up the configuration for developing your project locally:
+supabase init
+
+# With docker installed and running start up your supabase services locally
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:120]:
```suggestion
# With Docker installed and running, start up your Supabase services locally
```

File: apps/docs/pages/guides/integrations/snaplet.mdx [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+Your **target database** may already contain these schemas when restoring. The warnings just mean that when Snaplet attempted to restore these schemas (**auth, graphql,** etc) but was unable to drop the ones already existing in your **target database** (since the user is not an owner of these), and consequently it was not able to create any structure for them (since that structure still exists)
+
+Snaplet will still make sure to clear all data for tables in these schemas and restore data for each of these tables to what is in the snapshot. In other words, **these warnings do not mean that the restore failed, but rather show you what Snaplet tried to do.**
+
+If you aren't actually needing the data for some of these schemas, you can stop these warnings by excluding the schema from the captured snapshots. You can read more on how to do this [over here in our docs](https://docs.snaplet.dev/references/data-operations/exclude):
```
Comment (charislam) [inline apps/docs/pages/guides/integrations/snaplet.mdx:145]:
```suggestion
If you don't need the data for some of these schemas, you can stop these warnings by excluding the schema from the captured snapshots. You can read more on how to do this [over here in our docs](https://docs.snaplet.dev/references/data-operations/exclude):
```

**Comment by khaya-zulu** [comment]
Hi @charislam, thanks for the edit suggestions 🙂. I see that the `pages/guides/integrations` route has been removed, is there another other way to make a contribution to the Snaplet integration page?

Happy to close this pr and push to another repo or share a doc if its internal.

**Comment by kiwicopple** [comment]
closing this since we have a new system for integrations (moving to self-serve rather than a PR approach)

**Pull request closed without merging**